Tools/Scripts/webkitpy/xcode/simulated_device.py

```diff
@@ -189,7 +196,7 @@ def _find_exisiting_device_for_request(request):
                 if isinstance(initialized_device, Device) and device == initialized_device:
                     device = None
                     break
-            if device and request.device_type == device.device_type:
+            if device and request.device_type == device.device_type and not device.platform_device.is_booted_or_booting():
```

Comment:
so the states we have:

```python
    class DeviceState:
        CREATING = 0
        SHUT_DOWN = 1
        BOOTING = 2
        BOOTED = 3
        SHUTTING_DOWN = 4
```

Things in `SimulatedDeviceManager.INITIALIZED_DEVICES` that we skip should be in `BOOTED` (except I guess there's nothing stopping them from being in `SHUTTING_DOWN` or `SHUT_DOWN` too), and I'd have thought we didn't want to return anything in `CREATING` here either?

Like, I think we should only be returning things in the `SHUT_DOWN` state which *aren't* in `SimulatedDeviceManager.INITIALIZED_DEVICES` here, and this code still allows `CREATING` and `SHUTTING_DOWN` I think?